src/main/java/com/laytonsmith/core/functions/Function.java

```diff
@@ -74,6 +78,62 @@
 	 */
 	public Mixed exec(Target t, Environment environment, Mixed... args) throws ConfigRuntimeException;
 
+	/**
+	 * Gets the return type of this function, based on the types of the passed arguments.
+	 * @param t - The code target, used for setting the code target in thrown exceptions.
+	 * @param argTypes - The types of the passed arguments.
+	 * @param argTargets - The {@link Target}s belonging to the argTypes (in the same order).
+	 * @param env - The {@link Environment}, used for instanceof checks on types.
+	 * @param exceptions - A set to which all type errors will be added.
+	 * @return The return type of this function when invoked with the given argument types. If the return type
+	 * is unknown, {@link CClassType#AUTO} is returned, indicating that this value should be handled as dynamic
+	 * during static type checking.
+	 */
+	public CClassType getReturnType(Target t, List<CClassType> argTypes,
```

Comment:
Out of scope for this PR, but necessary before implementing typechecking in all functions (to prevent double work). Once there is some signature setup and all functions return some signature, I don't mind implementing some default implementation of getReturnType() to use this signature.